### PR TITLE
Specify return object in case key not found

### PR DESCRIPTION
**Pull request by persiaAziz**

### CODE REVIEW

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/303/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/316/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/299/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/428/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1890/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/300/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/429/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/1998/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1891/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/561/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/562/

File: lib/ts/Map.h [hunk 1]
```diff
@@ -114,11 +114,16 @@ template <class K, class AHashFns, class C, class A = DefaultAlloc> class HashMa
   using Map<K, C, A>::i;
   using Map<K, C, A>::v;
   using Map<K, C, A>::e;
+  HashMap() : defaultRet(0) {}
```
Comment (SolidWallOfCode) [inline lib/ts/Map.h]: No member initializer, it's done in the declaration.

File: lib/ts/Map.h [hunk 2]
```diff
@@ -114,11 +114,16 @@ template <class K, class AHashFns, class C, class A = DefaultAlloc> class HashMa
...
+  HashMap(C c) : defaultRet(c) {}
   MapElem<K, C> *get_internal(K akey);
   C get(K akey);
   value_type *put(K akey, C avalue);
   void get_keys(Vec<K> &keys);
   void get_values(Vec<C> &values);
+
+private:
+  C defaultRet = 0; // return this object if key is not present
```
Comment (SolidWallOfCode) [inline lib/ts/Map.h]: I think `invalid_value` would be a better name.

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/309/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/322/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1896/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/306/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/435/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2004/

**Comment by persiaAziz** [comment]
done @SolidWallOfCode

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/568/

**Comment by SolidWallOfCode** [comment]
Long term, we're going to rewrite this entire class and do it right. Until this, this is the best we can do because that rewrite is going to be a large, invasive project.

**Pull request merged**